File: SNM.FrontEnd/src/app/core/models/shared/auditableEntityBase.ts
import { IAuditableEntityBase } from "app/core/types/shared/auditableEntityBase.types";

export class  AuditableEntityBase implements Required<IAuditableEntityBase> {
    id: string | null;
    createdBy: string | null;
    createdOn: Date | null;
    lastModifiedBy: string | null;
    lastModifiedOn: string | null;
    deletedOn: string | null;
    deletedBy: string | null;

    //  constructor(entity: IAuditableEntityBase) {
    // //     this.id = entity.id || null;
    //     this.createdBy = entity.createdBy || null;
    //     this.createdOn = entity.createdOn || null;
    //     this.lastModifiedBy = entity.lastModifiedBy || null;
    //     this.lastModifiedOn = entity.lastModifiedOn || null;
    //     this.deletedOn = entity.deletedOn || null;
    //     this.deletedBy = entity.deletedBy || null;
    //  }
}

File: SNM.Twitter/Core/SNM.Twitter.Domain/Common/AuditableEntityBase.cs
namespace SNM.Twitter.Domain.Common
{
#nullable enable
    public abstract class AuditableEntityBase<Tid> : EntityBase<Tid>
    {
        //public DateTime Created { get; set; }

        //public string? CreatedBy { get; set; }

        //public DateTime LastModified { get; set; }

        //public string? LastModifiedBy { get; set; }
        public Guid CreatedBy { get; set; }
        public DateTime CreatedOn { get; private set; }
        public Guid LastModifiedBy { get; set; }
        public DateTime? LastModifiedOn { get; set; }
        public DateTime? DeletedOn { get; set; }
        public Guid? DeletedBy { get; set; }
    }
}